internal/overlord/servstate/handlers.go

```diff
@@ -349,7 +350,8 @@ func (s *serviceData) startInternal() error {
 		// started (previous logs have already been copied).
 		outputIterator = s.logs.HeadIterator(0)
 	}
-	logWriter := servicelog.NewFormatWriter(s.logs, s.config.Name)
+	formatWriter := servicelog.NewFormatWriter(s.logs, s.config.Name)
+	logWriter := servicelog.NewTrimWriter(formatWriter, regexp.MustCompile("^"+s.config.LogTrim))
```

Comment:
I think we need parentheses around this ... consider if the `log-trim` regex is `foo|bar`, we'd end up with `^foo|bar`, which [matches](https://go.dev/play/p/pQwTQIWa3a1) "foo" at the start of the line and "bar" anywhere in the line, because `^` is higher precedence than `|`. So we should do `"^("+s.config.LogTrim+")"`, or `"^(?:"+s.config.LogTrim+")"` if we need the non-capturing version (I don't think you're using submatches here, so probably the first version is fine).

Also, why are we doing this unconditionally now? I quite liked the fact that previously you only got the performance penalty if `log-trim` was set.

Reply:
That occurred to me.  2 thoughts:

* If we add default regexes, it seems like it will be rare for users to explicitly blank it for performance reasons.  So in practice the performance win would almost never be realized.  Although it would at least make the workaround possible...
* To avoid the writer wrap, we either need to add a Flush method to the format writer (easy) or a type assertion where we call Flush.

Seems like maybe just adding a Flush method to the regular formatter would be the cleanest

Reply:
I don't like adding a `Flush` method to the formatter if it's not needed. I think a type assertion where you call `Flush` is fine, like `if f, ok := writer.(interface{ Flush() }); ok { f.Flush() }`. Another approach I used recently was to use a func variable initialized to a no-op function, something like this:

```
flush := func() {}
if logTrim {
    logWriter := NewLogWriter(writer)
    writer = logWriter
    flush = func() { logWriter.Flush() }
}
...
flush()
```

Reply:
I like the inline func var - done.